com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext

```diff
@@ -122,87 +127,132 @@ Comparison:
 	AttributeComparison | DataTypeComparison;
 
 AttributeComparison:
-	AttributeValueEquals | AttributeValueNotEquals;
+	op=NON_NUMERIC_OPERATOR value=FilteredExpressionConstraint;
 
 DataTypeComparison:
-	BooleanValueEquals
-	| BooleanValueNotEquals
-	| StringValueEquals
-	| StringValueNotEquals
-	| IntegerValueEquals
-	| IntegerValueNotEquals
-	| IntegerValueGreaterThan
-	| IntegerValueGreaterThanEquals
-	| IntegerValueLessThan
-	| IntegerValueLessThanEquals
-	| DecimalValueEquals
-	| DecimalValueNotEquals
-	| DecimalValueGreaterThan
-	| DecimalValueGreaterThanEquals
-	| DecimalValueLessThan
-	| DecimalValueLessThanEquals;
+	BooleanValueComparison
+	| StringValueComparison
+	| IntegerValueComparison
+	| DecimalValueComparison;
 
-AttributeValueEquals:
-	EQUAL constraint=SubExpressionConstraint;
+BooleanValueComparison:
+	op=NON_NUMERIC_OPERATOR value=Boolean;
 
-AttributeValueNotEquals:
-	NOT_EQUAL constraint=SubExpressionConstraint;
+StringValueComparison:
+	op=NON_NUMERIC_OPERATOR value=STRING;
 
-BooleanValueEquals:
-	EQUAL value=Boolean;
+IntegerValueComparison:
+	op=NUMERIC_OPERATOR HASH value=Integer;
 
-BooleanValueNotEquals:
-	NOT_EQUAL value=Boolean;
+DecimalValueComparison:
+	op=NUMERIC_OPERATOR HASH value=Decimal;
 
-StringValueEquals:
-	EQUAL value=STRING;
+NestedExpression:
+	ROUND_OPEN nested=ExpressionConstraint ROUND_CLOSE;
 
-StringValueNotEquals:
-	NOT_EQUAL value=STRING;
+// filters
 
-IntegerValueEquals:
-	EQUAL HASH value=Integer;
+FilterConstraint:
+	DOUBLE_CURLY_OPEN Filter DOUBLE_CURLY_CLOSE;
 
-IntegerValueNotEquals:
-	NOT_EQUAL HASH value=Integer;
+Filter:
+	DisjunctionFilter;
 
-IntegerValueGreaterThan:
-	GT HASH value=Integer;
+// Conjunction (OR) of filters is an extension to ECL 1.5
+DisjunctionFilter returns Filter:
+	ConjunctionFilter ({DisjunctionFilter.left=current} DISJUNCTION right=ConjunctionFilter)*;
 
-IntegerValueLessThan:
-	LT HASH value=Integer;
+// Using the "AND" keyword for disjunctions is an extension to ECL 1.5
+ConjunctionFilter returns Filter:
+	PropertyFilter ({ConjunctionFilter.left=current} (CONJUNCTION | COMMA) right=PropertyFilter)*;
 
-IntegerValueGreaterThanEquals:
-	GTE HASH value=Integer;
+NestedFilter:
+	ROUND_OPEN nested=Filter ROUND_CLOSE;
 
-IntegerValueLessThanEquals:
-	LTE HASH value=Integer;
+PropertyFilter:
+	// ECL 1.5 filters
+	TermFilter 
+	| LanguageFilter 
+	| TypeFilter 
+	| DialectFilter 
+	// QL 0.1 filters (ECL extensions)	 
+	| ActiveFilter 
+	| PreferredInFilter
+	| AcceptableInFilter
+	| LanguageRefSetFilter
+	| ModuleFilter
+	| CaseSignificanceFilter 
```

Comment:
Could you please add the following filters as well? 🙇
* SemanticTagFilter (supported domains are `Concept` and `Description`)
* EffectiveTimeFilter (supported domains are `Concept` and `Description`)